### DISCUSSION TITLE
Publishing node_modules?

### DISCUSSION THREAD

**Discussion by m3Lith** [bikecoders/ngx-deploy-npm#534] [Q&A] [answered]
How would a setup for publishing a package and related `node_modules` look like?

Now I have an Nx workspace with a `package.json` in the root folder , which contains all the npm package dependencies.

The lib that I want to publish has no dependencies in it's `package.json` - they're taken from the root node_modules.
If I'm publishing the lib, `node_modules` are not being included.

**Comment by dianjuar** [comment, marked as answer]
That's a good question.
 
My experience with Nx tells me that **you need to list the dependencies of your library on the library's package.json**.

That doesn't mean that you will remove the dependencies from the workspace root. The dependencies on the library's package.json indicate NPM what it needs to install when installing your library. That process is automatically managed by npm when installing your library.

If you don't do that and you are using RxJs, for instance, what would happen is that in run time, your library will throw an exception indicating that RxJs was not found.

**Reply by dianjuar** [reply]
You can test that using `yalc` or publishing your library to a local registry like `verdaccio`.

That will allows you to simulate the whole process and observe how it works

**Reply by dianjuar** [reply]
Take an example of [@jscutlery/semver](https://github.com/jscutlery/semver/blob/main/package.json)

**Reply by dianjuar** [reply]
You can configure an EsLint to indicate when is time to update your package's dependencies.

https://dev.to/this-is-learning/manage-nx-library-dependencies-with-the-nxdependency-checks-eslint-rule-2lem
